src/hashes/sha3_test.c

```diff
@@ -388,161 +388,261 @@ int sha3_shake_test(void)
+      { 256, 17*17*17*17*17*17,     0, 64, "9e11bc59c24e73993c1484ec66358ef71db74aefd84e123f7800ba9c4853e02cfe701d9e6bb765a304f0dc34a4ee3ba82c410f0da70e86bfbd90ea877c2d6104" },
    };
+   for (counter = 0; counter < (int)LTC_ARRAY_SIZE(testcases); counter++) {
+      if ((err = s_turbo_shake_test_one(&testcases[counter], counter)) != CRYPT_OK) {
+         return err;
+      }
+   }
+   return CRYPT_OK;
+#endif
+}
+#endif
 
-   const unsigned char turbo_shake_256_empty[64] = {
-      0x36, 0x7a, 0x32, 0x9d, 0xaf, 0xea, 0x87, 0x1c, 0x78, 0x02, 0xec, 0x67, 0xf9, 0x05, 0xae, 0x13,
-      0xc5, 0x76, 0x95, 0xdc, 0x2c, 0x66, 0x63, 0xc6, 0x10, 0x35, 0xf5, 0x9a, 0x18, 0xf8, 0xe7, 0xdb,
-      0x11, 0xed, 0xc0, 0xe1, 0x2e, 0x91, 0xea, 0x60, 0xeb, 0x6b, 0x32, 0xdf, 0x06, 0xdd, 0x7f, 0x00,
-      0x2f, 0xba, 0xfa, 0xbb, 0x6e, 0x13, 0xec, 0x1c, 0xc2, 0x0d, 0x99, 0x55, 0x47, 0x60, 0x0d, 0xb0,
-   };
-   const unsigned char turbo_shake_256_empty_10032[32] = {
-      0xab, 0xef, 0xa1, 0x16, 0x30, 0xc6, 0x61, 0x26, 0x92, 0x49, 0x74, 0x26, 0x85, 0xec, 0x08, 0x2f,
-      0x20, 0x72, 0x65, 0xdc, 0xcf, 0x2f, 0x43, 0x53, 0x4e, 0x9c, 0x61, 0xba, 0x0c, 0x9d, 0x1d, 0x75,
-   };
-   const unsigned char turbo_shake_256_single_zero_byte[64] = {
-      0x3e, 0x17, 0x12, 0xf9, 0x28, 0xf8, 0xea, 0xf1, 0x05, 0x46, 0x32, 0xb2, 0xaa, 0x0a, 0x24, 0x6e,
-      0xd8, 0xb0, 0xc3, 0x78, 0x72, 0x8f, 0x60, 0xbc, 0x97, 0x04, 0x10, 0x15, 0x5c, 0x28, 0x82, 0x0e,
-      0x90, 0xcc, 0x90, 0xd8, 0xa3, 0x00, 0x6a, 0xa2, 0x37, 0x2c, 0x5c, 0x5e, 0xa1, 0x76, 0xb0, 0x68,
-      0x2b, 0xf2, 0x2b, 0xae, 0x74, 0x67, 0xac, 0x94, 0xf7, 0x4d, 0x43, 0xd3, 0x9b, 0x04, 0x82, 0xe2,
-   };
-   const unsigned char turbo_shake_256_ptn_pow_1[64] = {
-      0xb3, 0xba, 0xb0, 0x30, 0x0e, 0x6a, 0x19, 0x1f, 0xbe, 0x61, 0x37, 0x93, 0x98, 0x35, 0x92, 0x35,
-      0x78, 0x79, 0x4e, 0xa5, 0x48, 0x43, 0xf5, 0x01, 0x10, 0x90, 0xfa, 0x2f, 0x37, 0x80, 0xa9, 0xe5,
-      0xcb, 0x22, 0xc5, 0x9d, 0x78, 0xb4, 0x0a, 0x0f, 0xbf, 0xf9, 0xe6, 0x72, 0xc0, 0xfb, 0xe0, 0x97,
-      0x0b, 0xd2, 0xc8, 0x45, 0x09, 0x1c, 0x60, 0x44, 0xd6, 0x87, 0x05, 0x4d, 0xa5, 0xd8, 0xe9, 0xc7,
-   };
-   const unsigned char turbo_shake_256_ptn_pow_2[64] = {
-      0x66, 0xb8, 0x10, 0xdb, 0x8e, 0x90, 0x78, 0x04, 0x24, 0xc0, 0x84, 0x73, 0x72, 0xfd, 0xc9, 0x57,
-      0x10, 0x88, 0x2f, 0xde, 0x31, 0xc6, 0xdf, 0x75, 0xbe, 0xb9, 0xd4, 0xcd, 0x93, 0x05, 0xcf, 0xca,
-      0xe3, 0x5e, 0x7b, 0x83, 0xe8, 0xb7, 0xe6, 0xeb, 0x4b, 0x78, 0x60, 0x58, 0x80, 0x11, 0x63, 0x16,
-      0xfe, 0x2c, 0x07, 0x8a, 0x09, 0xb9, 0x4a, 0xd7, 0xb8, 0x21, 0x3c, 0x0a, 0x73, 0x8b, 0x65, 0xc0,
-   };
+#ifdef LTC_KANGAROO_TWELVE
+typedef struct kangaroo_twelve_test_s {
+   int bits_count, is_ptn;
+   unsigned long input_bytes_count, customization_bytes_count, skip_digest_bytes, digest_bytes_count;
+   const char *expected_digest_hex;
+} kangaroo_twelve_test_t;
```

Comment:
```suggestion
typedef struct kangaroo_twelve_test {
   int bits_count, is_ptn;
   unsigned long input_bytes_count, customization_bytes_count, skip_digest_bytes, digest_bytes_count;
   const char *expected_digest_hex;
} kangaroo_twelve_test;
```

If you typedef, always call the `struct` and the `typedef` the same, and please don't suffix if there's no need for it.

We've started to prefix with `ltc_` if they're declared in a header and if they're new infrastructure, e.g. the X.509 implementation that will be merged at one point.

Reply:
Can not do because `src/hashes/sha3_test.c:517:3: error: ‘kangaroo_twelve_test’ redeclared as different kind of symbol`, changed to `kangaroo_twelve_test_case` instead.